contrib/nydusify/pkg/optimizer/builder.go

```diff
@@ -0,0 +1,87 @@
+package optimizer
+
+import (
+	"context"
+	"encoding/json"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+var logger = logrus.WithField("module", "optimizer")
+
+func isSignalKilled(err error) bool {
+	return strings.Contains(err.Error(), "signal: killed")
+}
+
+type BuildOption struct {
+	BuilderPath         string
+	PrefetchFilesPath   string
+	BootstrapPath       string
+	BlobDir             string
+	OutputBootstrapPath string
+	OutputJSONPath      string
+	Timeout             *time.Duration
+}
+
+type outputJSON struct {
+	Blobs []string `json:"blobs"`
+}
+
+func Build(option BuildOption) (string, error) {
```

Comment:
`func Optimize`

Reply:
this conflicts `Optimize` declaration in optimizer.go